### PR TITLE
[feat] cred authentication for SMTP service

### PR DESCRIPTION
**Pull request by kalikaneko**

### CODE REVIEW

File: src/leap/mail/smtp/__init__.py [hunk 1]
```diff
@@ -26,44 +26,33 @@
 from leap.mail.outgoing.service import OutgoingMail
```
Comment (meskio) [inline src/leap/mail/smtp/__init__.py]: Not used

File: src/leap/mail/smtp/gateway.py [hunk 2]
```diff
@@ -245,6 +311,7 @@ def validateFrom(self, helo, origin):
         @raise twisted.mail.smtp.SMTPBadSender: Raised if messages from this
             address are not to be accepted.
         """
+        print "VALIDATE FROM 2", helo, origin
```
Comment (meskio) [inline src/leap/mail/smtp/gateway.py]: I guess this is print can be removed.

File: src/leap/mail/smtp/__init__.py (hunk 1 above)
Comment (kalikaneko) [inline src/leap/mail/smtp/__init__.py]: done

File: src/leap/mail/smtp/gateway.py (hunk 2 above)
Comment (kalikaneko) [inline src/leap/mail/smtp/gateway.py]: print 'done'

Comment (meskio) [inline src/leap/mail/smtp/gateway.py]: ???

Comment (kalikaneko) [inline src/leap/mail/smtp/gateway.py]: hmm I thought I had removed the print, but it seems I didn't. pushed again.

**Pull request merged**